projects/go-lib/src/lib/components/go-switch-toggle/go-switch-toggle.component.ts

```diff
@@ -21,9 +22,15 @@ export class GoSwitchToggleComponent implements OnInit {
 
   ngOnInit(): void {
     this.id = this.key || generateId(this.label, 'toggle');
+
+    if (this.disabled) {
+      this.control.disable();
+    }
```

Comment:
Yeah, you're right. I did try doing it that way at first, but since the `toggle()` method still gets called even if the hidden input is disabled, and since we manually change the control value in `toggle()`, just setting the input to disabled wouldn't prevent you from actually toggling the control value. However, that was before I added the check for `!this.disabled` in `toggle()`, and now that that's there I can remove this stuff from `ngOnInit` and just disable the input the way you suggest. So thanks for pointing that out 👍